modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/resources/META-INF/resources/js/legacy/alloyeditor_source.js

```diff
@@ -123,77 +123,60 @@ AUI.add(
 					const host = instance.get(STR_HOST);
 					const strings = instance.get(STRINGS);
 
-					let fullScreenDialog = instance._fullScreenDialog;
 					let fullScreenEditor = instance._fullScreenEditor;
 
-					if (fullScreenDialog) {
-						fullScreenEditor.set('value', host.getHTML());
-
-						fullScreenDialog.show();
-					}
-					else {
-						Liferay.Util.openModal({
-							buttons: [
-								{
-									label: strings.cancel,
-									onClick: () => {
-										fullScreenDialog.hide();
-									},
+					Liferay.Util.openModal({
+						bodyHTML: '<div id="fullScreenEditorBody"></div>',
+						buttons: [
+							{
+								label: strings.cancel,
+								onClick: ({processClose}) => {
+									processClose();
 								},
-								{
-									displayType: 'primary',
-									label: strings.done,
-									onClick: () => {
-										fullScreenDialog.hide();
-										instance._switchMode({
-											content:
-												fullScreenEditor.get('value'),
-										});
-									},
+							},
+							{
+								displayType: 'primary',
+								label: strings.done,
+								onClick: ({processClose}) => {
+									processClose();
+									instance._switchMode({
+										content: fullScreenEditor.get('value'),
+									});
 								},
-							],
-							className:
-								'lfr-fulscreen-source-editor-dialog modal-full-screen',
-							containerProps: {},
-							onOpen: ({container}) => {
-								fullScreenDialog = container;
-
-								Liferay.Util.getTop()
-									.AUI()
-									.use(
-										'liferay-fullscreen-source-editor',
-										(A) => {
-											fullScreenEditor =
-												new A.LiferayFullScreenSourceEditor(
-													{
-														boundingBox: container
-															.getStdModNode(
-																A.WidgetStdMod
-																	.BODY
-															)
-															.appendChild(
-																'<div></div>'
-															),
-														dataProcessor:
-															host.getNativeEditor()
-																.dataProcessor,
-														previewCssClass:
-															'alloy-editor alloy-editor-placeholder',
-														value: host.getHTML(),
-													}
-												).render();
-
-											instance._fullScreenDialog =
-												fullScreenDialog;
-
-											instance._fullScreenEditor =
-												fullScreenEditor;
-										}
-									);
 							},
-							title: strings.editContent,
-						});
-					}
+						],
+						className:
+							'lfr-fullscreen-source-editor-dialog modal-full-screen',
```

Comment:
Existing code is really odd to me. We branch on `if (fullScreenDialog) {`, but both paths lead to full screen dialog? I dunno. The code in PR makes sense to me, just simplifies it to one case.

Just one potential simplification here, can we use `size: 'full-screen'`? With that, you may not need CSS change in this PR.